### ISSUE TITLE
Last Used Ball isn't behaving like in the official games

### ISSUE THREAD

**Issue by LOuroboros** [rh-hideout/pokeemerald-expansion#4200]
### Description

ʟᴜɴᴏꜱ Today at 11:26 PM
Re: Last Used Ball
Shouldn't that be updated when I throw an Ultra Ball or w/e? Because it's always defaulting to a Poké Ball now.

It is updated normally in SwSh.
If you throw a Dusk Ball, the Last Used Ball selected will be a Dusk Ball.

It's also not saved inbetween battles at all.
If I throw a Master Ball, the next battle has a Poké Ball selected.

It is saved inbetween battles in SwSh.
If you use a Dive Ball to catch a Pokémon, a Dive Ball will be selected by default in the next battle.

https://github.com/rh-hideout/pokeemerald-expansion/assets/4485172/36ef7dfc-c297-44d7-91b9-8621945573a1

https://github.com/rh-hideout/pokeemerald-expansion/assets/4485172/7c0ba1df-e877-44e6-b1a7-6dcfaff7b196

### Version

upcoming (Edge)

### Upcoming/master Version

1f349e0fb9753f070bc6f0504b5ae1ea47fc1bc3

### Discord contact info

lunos4026

**Comment by WesleyJonker-Topicus** [comment]
Yesterday I fixed a bug on my own repo. When you run out of balls, obviously there is no Last Used Ball to show but when you buy new balls, the popup will not come back anymore. That seems to be because when you run out of balls, gBallToDisplay is set in TryAddLastUsedBallItemSprites. But not set to an actual item I believe. So I added a check to see if the ItemId to assign has a quantity in the bag with CheckBagHasItem. It seem to have fixed the issue for me.

Perhaps worth checking/fixing.

**Comment by Wesmaster** [comment]
Yesterday I fixed a bug on my own repo. When you run out of balls, obviously there is no Last Used Ball to show but when you buy new balls, the popup will not come back anymore. That seems to be because when you run out of balls, gBallToDisplay is set in TryAddLastUsedBallItemSprites. But not set to an actual item I believe. So I added a check to see if the ItemId to assign has a quantity in the bag with CheckBagHasItem. It seem to have fixed the issue for me.
